controllers/auth.controller.js

```diff
@@ -87,15 +75,28 @@ exports.validationLink = async (req, res, next) => {
 exports.verify = async (req, res, next) => {
   if (req.user.isVerified) {
     //if user is already verified return an error
-    return res.json(
-      sendResponse(httpStatus.BAD_REQUEST, 'The link has expired'),
-    );
+    return res.json(sendResponse(httpStatus.BAD_REQUEST, 'The link has expired'));
   }
   const user = req.user;
   user.isVerified = true;
   await user.save();
   const token = user.token();
-  return res.json(
-    sendResponse(httpStatus.OK, 'Verified', user.transform(), null, token),
-  );
+  return res.json(sendResponse(httpStatus.OK, 'Verified', user.transform(), null, token));
+};
+
+exports.changePassword = async (req, res, next) => {
+  const user = req.user;
+  const { password, newPassword } = req.body;
+
+  try {
+    if (!(await user.passwordMatches(password))) {
+      return res.json(sendResponse(httpStatus.NOT_FOUND, 'Invalid Password.'));
+    }
+    user.password = newPassword;
+    user.save();
```

Comment:
add an await to this.
```suggestion
    await user.save();
```